File: python/leetcode/question20.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack_s = []
        for charts in s:
            if charts in ['(', '{', '[']:
                stack_s.append(charts)
            else:
                if not stack_s:
                    return False
                if charts == ')' and stack_s[-1] == '(':
                    stack_s.pop()
                elif charts == ']' and stack_s[-1] == '[':
                    stack_s.pop()
                elif charts == '}' and stack_s[-1] == '{':
                    stack_s.pop()
                else:
                    return False
        return len(stack_s) == 0

File: fronted/typescript/leetcode/question20.ts
function isValid(s: string): boolean {
    const stack_s = [];
    const s_map = new Map([[')', '('],
    [']', '['],
    ['}', '{']])
    for (let charts of s) {
        if (!s_map.has(charts)) {
            stack_s.push(charts);
        } else {
            if (stack_s.length !== 0 && stack_s[stack_s.length - 1] === s_map.get(charts)) {
                stack_s.pop();
            } else {
                return false;
            }
        }
    }
    return stack_s.length === 0;
};